pkg/httpstream/spdy/upgrade.go

```diff
@@ -0,0 +1,127 @@
+package spdy
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/httpstream"
+	"github.com/docker/spdystream"
+)
+
+const (
+	headerConnection = "Connection"
+	headerUpgrade    = "Upgrade"
+	headerSpdy31     = "SPDY/3.1"
+)
+
+// connectionUpgrader knows how to upgrade HTTP requests and responses. It
+// implements the httpstream.ConnectionUpgrader interface.
+type connectionUpgrader struct {
+}
+
+// NewConnectionUpgrader returns a new httpstream.ConnectionUpgrader that is
+// capable of upgrading HTTP requests and responses using SPDY/3.1 via the
+// spdystream package.
+func NewConnectionUpgrader() httpstream.ConnectionUpgrader {
+	return connectionUpgrader{}
+}
+
+// UpgradeRequest upgrades an HTTP request to one that supports multiplexed
+// streams. newStreamHandler will be called synchronously whenever the
+// other end of the upgraded connection creates a new stream.
+//
+// UpgradeRequest uses req to create its own net.Connection to the remote
+// server.
+func (u connectionUpgrader) UpgradeRequest(req *http.Request, newStreamHandler httpstream.NewStreamHandler) (httpstream.Connection, error) {
+	req.Header.Add(headerConnection, headerUpgrade)
+	req.Header.Add(headerUpgrade, headerSpdy31)
+
+	// TODO ideally we'd tie in to the Kubernetes Client.Do() but there's no easy
```

Comment:
Do you have any suggestions how to plumb this together? In `pkg/kubectl/cmd/exec.go` I get a client.Client, and then use its RESTClient to create a Get request and then I call ExecuteRemoteCommand. I don't have access to the client.Config anywhere I can see. The TLS settings are in Client.RESTClient.Client.Transport.TLSClientConfig, but I can't walk that path because Client.RESTClient.Client is the HTTPClient interface, which just has Do(). Do you want me to do type assertions or did you have something else in mind?